src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java

```diff
@@ -45,95 +42,158 @@ public final class ThrowableToErrorMapper {
+          .getCommandResultError(
+              ErrorCode.UNAUTHENTICATED_REQUEST.getMessage(), Response.Status.UNAUTHORIZED);
+    } else if (message.contains(
+            "If you want to execute this query despite the performance unpredictability, use ALLOW FILTERING")
+        || message.contains("ANN ordering by vector requires the column to be indexed")) {
+      return ErrorCode.SERVER_NO_INDEX_ERROR
+          .toApiException()
+          .getCommandResultError(ErrorCode.SERVER_NO_INDEX_ERROR.getMessage(), Response.Status.OK);
+    }
+    String errorMessage =
+        message.contains("vector<float,") ? "Mismatched vector dimension" : message;
+    return ErrorCode.SERVER_INVALID_QUERY
+        .toApiException()
+        .getCommandResultError(errorMessage, Response.Status.OK);
+  }
+
+  /**
+   * Driver AllNodesFailedException a composite exception, peeling the errors from it
+   *
+   * @param throwable
+   * @param message
+   * @return
+   */
+  private static CommandResult.Error handleAllNodesFailedException(
+      AllNodesFailedException throwable,
+      String message,
+      Map<String, Object> fields,
+      Map<String, Object> fieldsForMetricsTag) {
+    Map<Node, List<Throwable>> nodewiseErrors = throwable.getAllErrors();
+    if (!nodewiseErrors.isEmpty()) {
+      List<Throwable> errors = nodewiseErrors.values().iterator().next();
+      if (errors != null && !errors.isEmpty()) {
+        Throwable error =
+            errors.stream()
+                .findAny()
+                .filter(
+                    t ->
+                        t instanceof AuthenticationException
+                            || t instanceof IllegalArgumentException
+                            || t instanceof NoNodeAvailableException)
+                .orElse(null);
+        // connect to oss cassandra throws AuthenticationException for invalid credentials
+        // connect to AstraDB throws IllegalArgumentException for invalid token/credentials
+        if (error instanceof AuthenticationException
+            || (error instanceof IllegalArgumentException
+                && (error.getMessage().contains("AUTHENTICATION ERROR")
+                    || error
+                        .getMessage()
+                        .contains("Provided username token and/or password are incorrect")))) {
+          return ErrorCode.UNAUTHENTICATED_REQUEST
+              .toApiException()
+              .getCommandResultError(
+                  ErrorCode.UNAUTHENTICATED_REQUEST.getMessage(), Response.Status.UNAUTHORIZED);
+          // Driver NoNodeAvailableException -> ErrorCode.NO_NODE_AVAILABLE
+        } else if (error instanceof NoNodeAvailableException) {
+          return ErrorCode.SERVER_NO_NODE_AVAILABLE
+              .toApiException()
+              .getCommandResultError(message, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+      }
+    }
+    return new CommandResult.Error(message, fieldsForMetricsTag, fields, Response.Status.OK);
```

Comment:
Same comment as above "Does this mean we are not returning a JsonApiException if we get to here? If it's an exception we don't recognize in any of our logic, should we really be returning HTTP OK?"